consumers/jira_c/gojira/jirautils.go

```diff
@@ -0,0 +1,162 @@
+package gojira
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/andygrunwald/go-jira"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	// EnvConfigPath the path towards the config.yaml file
+	EnvConfigPath = "DRACON_JIRA_CONFIG_PATH"
+)
+
+type customField struct {
+	ID        string   `yaml:"id"`
+	FieldType string   `yaml:"fieldType"`
+	Values    []string `yaml:"values"`
+}
+
+type defaultValues struct {
+	IssueFields  map[string][]string `yaml:"issueFields,omitempty"`
+	CustomFields []customField       `yaml:"customFields,omitempty"`
+}
+
+type mappings struct {
+	DraconField string `yaml:"draconField"`
+	JiraField   string `yaml:"jiraField"`
+	FieldType   string `yaml:"fieldType"`
+}
+
+type config struct {
+	DefaultValues     defaultValues `yaml:"defaultValues"`
+	Mappings          []mappings    `yaml:"mappings"`
+	DescriptionExtras []string      `yaml:"addToDescription"`
+}
+
+// getConfig parses the configuration from config.yaml and returns a config struct
+func getConfig() config {
+	configFile, err := ioutil.ReadFile(os.Getenv(EnvConfigPath))
+	if err != nil {
+		log.Printf("Error while reading config file:   #%v ", err)
+	}
+
+	var newConfig config
+	err = yaml.Unmarshal(configFile, &newConfig)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+	return newConfig
+}
+
+// parseDraconMessage returns a hashmap of the parsed dracon message
+func parseDraconMessage(message string) (map[string]string, error) {
+	jBytes := []byte(message)
+	var draconResult map[string]string
+	err := json.Unmarshal(jBytes, &draconResult)
+	return draconResult, err
+}
+
+// makeCustomField returns the appropriate interface for a jira CustomField given it's type and values
+// :param fieldType: the type of the field in Jira (single-value, multi-value, float)
+// :param values: list of values to be filled in
+// :return the appropriate interface for a CustomField, given the corresponding fieldType and value(s)
+func makeCustomField(fieldType string, values []string) interface{} {
```

Comment:
what's the impact of this returning `nil` if a field type isn't supported? do you think it's worth returning an error if a field isn't supported?

Reply:
I've tested this and the code doesn't crash, it just results in an empty Field. (Exception is if the Field has a set of supported values and `nil` is not one of them -- but this is not something that you would be able to check in the code; createIssue() will throw you this error later)

So I've added a Warning message if the fieldType isn't supported, but the code can still run.